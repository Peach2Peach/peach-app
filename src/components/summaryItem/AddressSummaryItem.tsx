import { View } from "react-native";
import tw from "../../styles/tailwind";
import { ShortBitcoinAddress } from "../bitcoin/ShortBitcoinAddress";
import { PeachText } from "../text/PeachText";
import { CopyAble } from "../ui/CopyAble";
import { SummaryItem, SummaryItemProps } from "./SummaryItem";
import { tolgee } from "../../tolgee";

type Props = SummaryItemProps & {
  address?: string;
};

export const AddressSummaryItem = ({ address, ...props }: Props) => (
  <SummaryItem {...props}>
    <View style={tw`flex-row items-center gap-2`}>
      {address ? (
        <ShortBitcoinAddress style={tw`subtitle-1`} address={address} />
      ) : (
        <PeachText style={tw`subtitle-1`}>
          {tolgee.t("loading", { ns: "unassigned" })}
        </PeachText>
      )}
      <CopyAble value={address} />
    </View>
  </SummaryItem>
);
