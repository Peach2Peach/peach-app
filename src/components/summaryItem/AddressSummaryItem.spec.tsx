import { createRenderer } from 'react-test-renderer/shallow'
import { AddressSummaryItem } from './AddressSummaryItem'

describe('AddressSummaryItem', () => {
  const renderer = createRenderer()
  it('renders correctly', () => {
    renderer.render(
      <AddressSummaryItem title="to" address="bcrt1q70z7vw93cxs6jx7nav9cmcn5qvlv362qfudnqmz9fnk2hjvz5nus4c0fuh" />,
    )
    expect(renderer.getRenderOutput()).toMatchSnapshot()
  })
  it('renders correctly when no address is passed', () => {
    renderer.render(<AddressSummaryItem title="to" />)
    expect(renderer.getRenderOutput()).toMatchSnapshot()
  })
})
