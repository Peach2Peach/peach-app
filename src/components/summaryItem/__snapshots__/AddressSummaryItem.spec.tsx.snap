// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddressSummaryItem renders correctly 1`] = `
<SummaryItem
  title="to"
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "gap": 8,
      }
    }
  >
    <ShortBitcoinAddress
      address="bcrt1q70z7vw93cxs6jx7nav9cmcn5qvlv362qfudnqmz9fnk2hjvz5nus4c0fuh"
      style={
        {
          "fontFamily": "Baloo2-SemiBold",
          "fontSize": 16,
          "letterSpacing": 0.16,
          "lineHeight": 28,
        }
      }
    />
    <CopyAble
      value="bcrt1q70z7vw93cxs6jx7nav9cmcn5qvlv362qfudnqmz9fnk2hjvz5nus4c0fuh"
    />
  </View>
</SummaryItem>
`;
